src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs

```diff
@@ -0,0 +1,155 @@
+
+namespace Bicep.LanguageServer.Handlers
+{
+    [Method(BicepDeploymentScopeRequestHandler.MethodName, Direction.ClientToServer)]
+    public record BicepDeploymentScopeParams(TextDocumentIdentifier TextDocument) : ITextDocumentIdentifierParams, IRequest<BicepDeploymentScopeResponse>;
+
+    public record BicepDeploymentScopeResponse(string scope, string? template, string? errorMessage);
+
+    /// <summary>
+    /// Handles bicep/getDeploymentScope LSP request.
+    /// The BicepDeploymentScopeRequestHandler returns targetScope, template and error message.
+    /// Error message would be null if provided bicep file was error free.
+    /// </summary>
+    public class BicepDeploymentScopeRequestHandler : IJsonRpcRequestHandler<BicepDeploymentScopeParams, BicepDeploymentScopeResponse>
+    {
+        public const string MethodName = "bicep/getDeploymentScope";
+
+        private readonly EmitterSettings emitterSettings;
+        private readonly ICompilationManager compilationManager;
+        private readonly IConfigurationManager configurationManager;
+        private readonly IFeatureProvider features;
+        private readonly IFileResolver fileResolver;
+        private readonly IModuleDispatcher moduleDispatcher;
+        private readonly INamespaceProvider namespaceProvider;
+
+        public BicepDeploymentScopeRequestHandler(
+            EmitterSettings emitterSettings,
+            ICompilationManager compilationManager,
+            IConfigurationManager configurationManager,
+            IFeatureProvider features,
+            IFileResolver fileResolver,
+            IModuleDispatcher moduleDispatcher,
+            INamespaceProvider namespaceProvider)
+        {
+            this.compilationManager = compilationManager;
+            this.configurationManager = configurationManager;
+            this.emitterSettings = emitterSettings;
+            this.features = features;
+            this.fileResolver = fileResolver;
+            this.moduleDispatcher = moduleDispatcher;
+            this.namespaceProvider = namespaceProvider;
+        }
+
+        public Task<BicepDeploymentScopeResponse> Handle(BicepDeploymentScopeParams request, CancellationToken cancellationToken)
+        {
+            var documentUri = request.TextDocument.Uri;
+
+            Compilation? compilation;
+
+            try
+            {
+                compilation = GetCompilation(documentUri);
+                var deploymentScope = GetDeploymentScope(compilation.GetEntrypointSemanticModel().TargetScope);
+
+                return Task.FromResult(new BicepDeploymentScopeResponse(deploymentScope, GetCompiledFile(compilation, documentUri), null));
+            }
+            catch (Exception exception)
+            {
+                return Task.FromResult(new BicepDeploymentScopeResponse(ResourceScope.None.ToString(), null, exception.Message));
+            }
+        }
+
+        private static string GetDeploymentScope(ResourceScope resourceScope)
```

Comment:
[nit] We can convert the switch statement to a [switch expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression) introduced in C# 8.

Reply:
Updated here:
https://github.com/Azure/bicep/commit/008cdcff0bd17d4798adc47ce6dfc485cf41e0c2

Thanks!